### PR TITLE
Docker activity readme fix

### PR DESCRIPTION
**Pull request by NadeeshDilanga**
13th July commit has README and code review fixes

### CODE REVIEW

File: taverna-docker-activity/README.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+########################## Docker Activity Plugin ############################
+
+This module implements the taverna activity plugin for Docker. The plugin capable of following functionality.
```
Comment (alaninmcr) [inline taverna-docker-activity/README.txt:3]: plugin is

File: taverna-docker-activity/README.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+1. Inspect Image
+2. Create Container
+3. Start Container
+4. List Containers
+5. Stop Container  - TODO
+6. Delete Container - TODO
+
+Test module src/test/java/ covers each and every capability. org.apache.taverna.activities.docker.RemoteClient
+capable of executing docker commands over TCP on the actual remote docker host using the docker-java api.
```
Comment (alaninmcr) [inline taverna-docker-activity/README.txt:13]: is capable

File: taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/TestDockerCommands.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
     private static final String DOCKER_LOGIN_SUCCESS = "Login Succeeded";
 
-    public TestDockerCommands(){
+    public TestDockerCommands() {
         DockerContainerConfigurationImpl containerConfiguration = new DockerContainerConfigurationImpl(new TestConfigurationManager());
         containerConfiguration.getInternalPropertyMap().put(DockerContainerConfigurationImpl.NAME,CONTAINER_NAME);
```
Comment (alaninmcr) [inline taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/TestDockerCommands.java]: containerConfiguration should be a DockerContainerConfiguration. Any code specific to the implementation should not be visible

File: taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java [hunk 4]
```diff
@@ -52,6 +43,7 @@
 
     private DockerContainerConfigurationImpl containerConfiguration;
```
Comment (alaninmcr) [inline taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java]: Should be a DockerContainerConfiguration

File: taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java [hunk 5]
```diff
@@ -44,13 +49,17 @@
...
```
Comment (alaninmcr) [inline taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java]: Should be a DockerContainerConfiguration

File: taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java [hunk 6]
```diff
@@ -71,7 +80,8 @@ public void setup() throws Exception {
 	 */
 	@Test
 	public void testInspectImage() throws Exception {
-		DockerActivity activity = new DockerActivity(containerConfiguration);
+        containerConfiguration.getInternalPropertyMap().put(DockerContainerConfigurationImpl.IMAGE,IMAGE_NAME);
```
Comment (alaninmcr) [inline taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java]: Should not access implementation

File: taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java [hunk 7]
```diff
@@ -85,4 +95,57 @@ public void testInspectImage() throws Exception {
...
+        Map<String,Object> inputs = new HashMap<String,Object>();
+        inputs.put(DockerActivity.ACTION, DockerActivity.CREATE_CONTAINER);
+
+        Map<String, Class<?>> expectedOutputs = new HashMap<String, Class<?>>();
+        expectedOutputs.put(DockerActivity.RESPONSE_BODY_KEY, String.class);
+
+        Map<String,Object> outputs = ActivityInvoker.invokeAsyncActivity(activity, inputs, expectedOutputs);
+        System.out.println(outputs.get(DockerActivity.RESPONSE_BODY_KEY));
+        Assert.assertNotNull(outputs.get(DockerActivity.RESPONSE_BODY_KEY));
+        String id = new ObjectMapper().readTree((String)outputs.get(DockerActivity.RESPONSE_BODY_KEY)).get("container-id").textValue();
```
Comment (alaninmcr) [inline taverna-docker-activity/src/test/java/org/apache/taverna/activities/docker/test/DockerActivityTest.java]: Better split into several statements? Difficult to see what it is doing

**Comment by NadeeshDilanga** [comment]
Removed all places of DockerContainerConfigurationImpl and instead use DockerContainerConfiguration interface. Will send a pull request soon.

**Pull request merged**